src/preferences.rs

```diff
@@ -62,57 +153,51 @@ pub trait Preferences {
 }
 
 pub struct PreferencesImpl {
-    pub config: RwLock<MyConfig>,
+    pub dao: PreferencesDao,
 }
 
 impl Preferences for PreferencesImpl {
-    fn last_completed_reports_interval(&self, key: PreferencesKey) -> Option<ReportsInterval> {
-        match key {
-            PreferencesKey::LastCompletedReportsInterval => {
-                self.config.read().last_completed_reports_interval
-            }
-        }
+    fn last_completed_reports_interval(&self) -> Option<ReportsInterval> {
+        let str = self.dao.load("last_completed_reports_interval");
+        str.map(|str| {
+            let res = serde_json::from_str(str.as_ref());
+            expect_log!(res, "Invalid interval str")
+        })
     }
 
-    fn set_last_completed_reports_interval(&self, _key: PreferencesKey, value: ReportsInterval) {
-        let mut config = self.config.write();
-        config.last_completed_reports_interval = Some(value);
-
-        let res = confy::store("myprefs", *config);
-
-        if let Err(error) = res {
-            error!("Error storing preferences: {:?}", error)
-        }
+    fn set_last_completed_reports_interval(&self, value: ReportsInterval) {
+        let res = serde_json::to_string(&value);
```

Comment:
@i-schuetz Why conversion to string?

Reply:
Because the sqlite preferences table is (text, text)

Reply:
Another question would be why specifically to JSON. Honestly, it's the only to-string serialization we've at the moment in the app and I copied it. I doubt there's a difference with others for this small data type. Confy required values to implement `serde::ser::Serialize`, though I've not looked what kind of serialization it uses.

Reply:
Maybe we can use a binary serializer and store as blob. Looks like something we can come back to later. We definitely have to keep an eye on the preferences performance, as we may not always access them from background threads (though I consider it unlikely that we will call FFI from the main thread).